src/error.ts

```diff
@@ -50,8 +65,8 @@ export class UnknownError extends Error {
     Error.captureStackTrace(this, this.constructor);
   }
 
-  toJSON() {
-    return Object.assign(super.toJSON(), this);
+  toJSON(): ErrorJSON {
+    return { ...this, ...super.toJSON() };
```

Comment:
Imho this should be first and I would bet it caused unexpected behaviour on serialized properties. You called super.toJSON() that would for example serialize the suberrors, but then this object assign would pass them here unserialized.